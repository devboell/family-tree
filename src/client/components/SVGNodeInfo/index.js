import React from 'react'
import PropTypes from 'prop-types'
import PersonInfoPanel from './PersonInfoPanel'

const SVGNodeInfo = ({
  x,
  y,
  boxWidth,
  boxHeight,
  info,
}) =>
  <foreignObject
    x={x}
    y={y}
    width={boxWidth}
    height={boxHeight}
  >
    <PersonInfoPanel
      width={boxWidth}
      height={boxHeight}
    >
      <p>{info}</p>
    </PersonInfoPanel>
  </foreignObject>

SVGNodeInfo.propTypes = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  boxWidth: PropTypes.number.isRequired,
  boxHeight: PropTypes.number.isRequired,
  info: PropTypes.string.isRequired,
}


export default SVGNodeInfo
