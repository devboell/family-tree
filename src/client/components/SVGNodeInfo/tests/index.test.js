import React from 'react'
import SVGNodeInfo from '../index'

const props = {
  x: 200,
  y: 300,
  boxWidth: 80,
  boxHeight: 80,
  info: 'John',
}


it('SVGNodeInfo, renders component', () => {
  const wrapper = shallow(<SVGNodeInfo {...props} />)
  expect(wrapper).toMatchSnapshot()
})
