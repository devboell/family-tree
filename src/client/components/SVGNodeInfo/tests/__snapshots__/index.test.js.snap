// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGNodeInfo, renders component 1`] = `
<foreignObject
  height={80}
  width={80}
  x={200}
  y={300}
>
  <styled.div
    height={80}
    width={80}
  >
    <p>
      John
    </p>
  </styled.div>
</foreignObject>
`;
